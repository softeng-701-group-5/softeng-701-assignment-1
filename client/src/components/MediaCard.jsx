import React from 'react';
import {
  Card,
  CardHeader,
  CardMedia,
  CardContent,
  CardActions,
  Avatar,
  Typography,
  makeStyles,
} from '@material-ui/core';

import TwitterIcon from './../assets/twitter-icon.svg';
import SpotifyIcon from './../assets/spotify-icon.svg';
import GithubIcon from './../assets/github-icon.svg';
import RedditIcon from './../assets/reddit-icon.svg';

const useStyles = makeStyles(theme => ({
  root: {
    maxWidth: 345,
    padding: 0,
    margin: 0,
    border: 0,
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
  },
  media: {
    height: 0,
    paddingTop: '56.25%', // 16:9
  },
  avatar: {
    backgroundColor: '#d9d9d9',
  },
}));

export const MediaCard = props => {
  const classes = useStyles();
  /* Alters colour bar based on which media is passed in
   * colours and logos:
   * reddit:  #FF4500 https://redditupvoted.files.wordpress.com/2015/10/reddit_icon_twitter_fb.png
   * spotify: #23D05F https://pbs.twimg.com/profile_images/558366562424332288/8ObpK74F.png
   * github:  #010101 https://avatars0.githubusercontent.com/u/9919?s=280&v=4
   * twitter: #05ACF0 https://pmcdeadline2.files.wordpress.com/2016/09/twitter-logo.jpg
   */
  const barColour = param => {
    switch (param) {
      case 'reddit':
        return '#FF4500';
      case 'spotify':
        return '#23D05F';
      case 'github':
        return '#010101';
      case 'twitter':
        return '#05ACF0';
      default:
        return '#b3b3b3';
    }
  };

  const mediaIcon = param => {
    switch (param) {
      case 'reddit':
        return RedditIcon;
      case 'spotify':
        return SpotifyIcon;
      case 'github':
        return GithubIcon;
      case 'twitter':
        return TwitterIcon;
      default:
        return '';
    }
  };

  return (
    <Card className={classes.root}>
      <CardHeader
        avatar={
          <Avatar
            alt={props.username}
            src={
              props.avatarLink ||
              'https://img.icons8.com/windows/64/000000/user.png'
            }
            className={classes.avatar}
          />
        }
        title={props.title}
        subheader={`${props.username} - ${props.relativeTime}`}
      />

      {props.imageLink && (
        <CardMedia className={classes.media} image={props.imageLink} />
      )}

      <CardContent>
        <Typography variant="body2" color="textSecondary" component="p">
          {props.mainText}
        </Typography>
      </CardContent>

      <CardActions
        style={{ backgroundColor: barColour(props.media) }}
        disableSpacing
      >
        <img src={mediaIcon(props.media)} alt={`${props.media} logo`} />
      </CardActions>
    </Card>
  );
};
