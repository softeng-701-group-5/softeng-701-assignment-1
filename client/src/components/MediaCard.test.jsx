import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import { MediaCard } from './MediaCard';
import renderer from 'react-test-renderer';

describe('<MediaCard />', () => {
  let subject;

  beforeEach(() => {
    subject = mount(<MediaCard />);
  });

  afterEach(() => {
    subject.unmount();
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<MediaCard />, div);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<MediaCard />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
