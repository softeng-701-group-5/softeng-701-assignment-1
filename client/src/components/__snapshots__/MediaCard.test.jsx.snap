// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MediaCard /> renders correctly 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardHeader-root"
  >
    <div
      className="MuiCardHeader-avatar"
    >
      <div
        className="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-5"
      >
        <img
          className="MuiAvatar-img"
          src="https://img.icons8.com/windows/64/000000/user.png"
        />
      </div>
    </div>
    <div
      className="MuiCardHeader-content"
    >
      <span
        className="MuiTypography-root MuiCardHeader-subheader MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        undefined - undefined
      </span>
    </div>
    <div
      className="MuiCardHeader-action"
    >
      <button
        aria-label="settings"
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
  <div
    className="MuiCardContent-root"
  >
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
    />
  </div>
  <div
    className="MuiCardActions-root"
    style={
      Object {
        "backgroundColor": "#b3b3b3",
      }
    }
  >
    <img
      src=""
    />
  </div>
</div>
`;
